pyrs/core/summary_generator_stress.py

```diff
@@ -0,0 +1,200 @@
+        """
+        # do file name checks
+        if not filename:
+            raise RuntimeError('Failed to supply output filename for Stress CSV output')
+
+        self._error_postfix = ' when creating stress file: ' + str(filename) + '\n'
+
+        if not filename.endswith('.csv'):
+            raise RuntimeError('File name must end with extension ".csv"' + self._error_postfix)
+
+        # check for length of lists
+        self._filename: str = str(filename)
+
+        if not stress:
+            raise RuntimeError(
+                'Error: stress of type StressField input can\'t be invalid' + self._error_postfix)
+
+        self._stress = stress
+
+        # check for filenames in StrainField per direction
+        for direction in SummaryGeneratorStress.directions:
+            strain = self._get_strain_field(direction)
+            assert isinstance(strain, StrainField)
+
+            # add exception if filenames is empty
+            if not strain.filenames:
+                raise RuntimeError('StrainField filenames in direction ' + str(direction) +
+                                   'can\'t be empty for Stress CSV output ' + self._filename)
+
+    def _write_csv_header(self, handle):
+        """
+          write projects names, peak_tags, Young's modulus and Poisson ratio
+        """
+        header = ''
+
+        for direction in SummaryGeneratorStress.directions:
+            strain = self._get_strain_field(direction)
+            assert isinstance(strain, StrainField)
+
+            line = '# Direction ' + str(direction) + ': '
+            for filename in strain.filenames:
+                run_number = filename[filename.index('HB2B_') + 5: filename.index('.h5')]
+                line += str(run_number) + ', '
+
+            line = line[:-2] + '\n'
+            header += line
+
+        header += '# E: ' + str(self._stress.youngs_modulus) + '\n'
+        header += '# v: ' + str(self._stress.poisson_ratio) + '\n'
+
+        handle.write(header)
+
+    def write_summary_csv(self):
+        """
+            Public function to generate a summary csv file for stress and input fields
+        """
+        def _write_summary_csv_column_names(handle):
+            column_names = 'vx, vy, vz, d0, d0_error '
+            # directional variables
+            for field_3dir in SummaryGeneratorStress.fields_3dir:
+                for direction in SummaryGeneratorStress.directions:
+                    column_names += field_3dir + '_Dir' + direction + ', '
+                    column_names += field_3dir + '_Dir' + direction + '_error, '
+
+            column_names = column_names[:-2] + '\n'
+            handle.write(column_names)
+            return
+
+        def _write_summary_csv_body(handle):
+
+            def _write_number(number) -> str:
+                if math.isnan(number):
+                    return ', '
+                return str(number) + ', '
+
+            def _write_field_3d(row: int, field: str):
```

Comment:
missing docstring. Can you insert a one-liner description and also the meaning of the input parameters?